### PR TITLE
Release 1.2.6

### PR DESCRIPTION
**Pull request by KGuillemot**
### Added
 - **Filebeat** : Experimental support for filebeat. Inputs and Modules are working, output is hardcoded to internal's redis so a REDIS listener may pop logs and send them trought rsyslog / darwin pipelines.

 - **Listeners** : Add CarbonBlack events collect via API support (with listeners LOG-API)

### CODE REVIEW

File: vulture_os/services/frontend/form.py [hunk 1]
```diff
@@ -486,6 +497,13 @@ def clean(self):
                 self.add_error('ruleset', "This field is required.")
             if not cleaned_data.get('listening_mode'):
                 self.add_error('listening_mode', "This field is required.")
+        if mode == "filebeat":
+            # Filebeat always send JSON logs on internal redis
+            cleaned_data['ruleset'] = "generic_json"
```
Comment (frikilax) [inline vulture_os/services/frontend/form.py:502]:
```suggestion
            #  cleaned_data['ruleset'] = "generic_json"
```

File: vulture_os/services/frontend/models.py [hunk 2]
```diff
@@ -112,6 +114,40 @@
...
 )
 
+
+# Filebeat module list
+FILEBEAT_MODULE_PATH = "/usr/local/etc/filebeat/modules.d"
+FILEBEAT_MODULE_LIST = [('_custom', 'Custom Filebeat config')]
+FILEBEAT_MODULE_CONFIG = {'_custom': '# Be sure to select the appropriate "Filebeat listening mode"\n\
+# Use the following variables when needed: \n\n\
+# - %ip% : The IP to listen on\n\
+# - %port% : The PORT to listen on\n\n\
```
Comment (frikilax) [inline vulture_os/services/frontend/models.py:124]:
```suggestion
# - %ip% : This keyword will be replaced by your Listener's IP automatically\n\
# - %port% : This keyword will be replaced by your Listener's port automatically\n\n\
```

File: vulture_os/services/frontend/views.py [hunk 3]
```diff
@@ -403,11 +462,34 @@ def render_form(front, **kwargs):
...
-                """ If it is an Rsyslog only conf """
+
+                elif frontend.mode == "filebeat" and "filebeat_config" in changed_data and old_filebeat_filename:
+
+                    # API request deletion of rsyslog frontend filename
+                    Cluster.api_request('services.filebeat.filebeat.delete_conf', old_filebeat_filename)
+                    logger.info("Filebeat config '{}' deletion asked.".format(old_filebeat_filename))
+                
+
+                """ If it is a Rsyslog only config """
```
Comment (frikilax) [inline vulture_os/services/frontend/views.py:473]: useless (services.filebeat.filebeat.delete_conf does absolutely nothing)

File: vulture_os/services/rsyslogd/rsyslog.py [hunk 4]
```diff
@@ -59,13 +59,13 @@
...
 class RsyslogService(Service):
-    """ HAProxy service class wrapper """
+    """ Rsyslog service class wrapper """
 
     def __init__(self):
         super().__init__()
         self.model = RsyslogSettings
         self.service_name = "rsyslogd"
-        self.friendly_name = "Logging"
+        self.friendly_name = "R-Logging"
```
Comment (frikilax) [inline vulture_os/services/rsyslogd/rsyslog.py:68]:
```suggestion
        self.friendly_name = "Rsyslog Logging"
```

File: vulture_os/services/filebeat/filebeat.py [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+FILEBEAT_PERMS = "640"
+
+class FilebeatService(Service):
+    """ Filebeat service class wrapper """
+
+    def __init__(self):
+        super().__init__()
+        self.model = FilebeatSettings
+        self.service_name = "filebeat"
+        self.friendly_name = "F-Logging"
```
Comment (frikilax) [inline vulture_os/services/filebeat/filebeat.py:67]:
```suggestion
        self.friendly_name = "Filebeat Logging"
```

File: vulture_os/toolkit/api_parser/carbon_black/carbon_black.py [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+        while found <= available:
+
+            response = self.get_logs(found, since)
+
+            # Downloading may take some while, so refresh token in Redis
+            self.update_lock()
+
+            logs = response['results']
+            found = int(response['num_found']) + 1
+            available = int(response['num_found'])
```
Comment (frikilax) [inline vulture_os/toolkit/api_parser/carbon_black/carbon_black.py:168]: Soomething wrong here, 'found' and 'available' will never be useful

Comment (KGuillemot) [reply vulture_os/toolkit/api_parser/carbon_black/carbon_black.py]: Yes you're right. I fixed this loop according to documentation.
num_found can be higher than num_available.

**Review by frikilax — APPROVED** [review]

**Pull request merged**